File: 프로그래머스/2/42584. 주식가격/주식가격.java
class Solution {
    public int[] solution(int[] prices) {
        int length = prices.length;
        int[] answer = new int[length];
        
        for (int i = 0; i < length; i++) {
            int cnt = 0;
            int price = prices[i];
            for (int j = i + 1; j < length; j++) {
                int nextPrice = prices[j];
                if (price > nextPrice && i != length -1) {
                    cnt += 1;
                    break;
                }
                if (price <= nextPrice) {
                    cnt += 1;
                }
            }
            answer[i] = cnt;
        }
        return answer;
    }
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
from collections import deque

def solution(prices):
    answer = [0] * len(prices)
    
    q = deque(prices)
    
    index = 0
    while q:
        target = q.popleft()
        count = 0
        for next in q:
            count += 1
            if next < target:
                break
        answer[index] = count
        index += 1
    return answer
